queries/lua/locals.scm

```diff
@@ -29,6 +29,11 @@
    (identifier) @definition.function) @scope)
 (function_definition) @scope
 
+(local_variable_declaration
+  (variable_declarator
+    (identifier) @definition.function)
+  (function_definition) @scope)
```

Comment:
Isn't that @scope already mentioned earlier?

Reply:
It is helpful for my use case and, more importantly, in line with the other queries in locals.scm to have the `@definition.function` and `@scope` defined within the same query where possible.  This lets you link the different captures to each other e.g. this scope goes with this function.

But I could put my new query before the standalone `(function_definition) @scope` query if that's what you were thinking?

Reply:
I see - the issue is that the `function_definition` scope would show up twice since it would match two queries.  I was under the impression treesitter queries had precedence rules, but I am doubting that after reading `:h nvim-treesitter-query-extensions`. I'll use an alternation to get rid of the duplicate scope.

Reply:
I see - the issue is that the `function_definition` scope would show up twice since it would match two queries.  I was under the impression treesitter queries had precedence rules, but I am doubting that after reading `:h nvim-treesitter-query-extensions`. I'll use an alternation to get rid of the duplicate scope.

Reply:
sorry, `(function_definition) @scope` was later. I don't know whether capturing a node twice is bad or not. @vigoux is the maintainer for Lua.

Reply:
Capturing a node twice is not really good for locals, for highlights it is actually fine though.